ods_ci/tests/Resources/Page/ModelRegistry/ModelRegistry.resource

```diff
@@ -0,0 +1,400 @@
+    Set Suite Variable    ${DOMAIN}    ${domain}
+    Set Suite Variable    ${TOKEN}    ${token}
+    Log    Domain: ${DOMAIN}
+
+Generate ModelRegistry Certificates
+    [Documentation]    Generates OpenSSL certificates for Model-Registry using the generate_certs.sh script.
+    ${output}=    Run Process    pwd    shell=True    stdout=PIPE
+    Log    Bash Command Output: ${output.stdout}
+    ${generate_certs_script}=    Set Variable    tests/Resources/CLI/ModelRegistry/generate_certs.sh
+    ${certs_dir}=    Set Variable    certs
+    ${cert_files}=    Create List
+    ...    ${certs_dir}/domain.crt    ${certs_dir}/domain.key    ${certs_dir}/model-registry-db.crt
+    ...    ${certs_dir}/model-registry-db.csr    ${certs_dir}/model-registry-db.key
+    ...    ${certs_dir}/modelregistry-sample-grpc.domain.crt    ${certs_dir}/modelregistry-sample-grpc.domain.csr
+    ...    ${certs_dir}/modelregistry-sample-grpc.domain.ext    ${certs_dir}/modelregistry-sample-grpc.domain.key
+    ...    ${certs_dir}/modelregistry-sample-rest.domain.crt    ${certs_dir}/modelregistry-sample-rest.domain.csr
+    ...    ${certs_dir}/modelregistry-sample-rest.domain.ext    ${certs_dir}/modelregistry-sample-rest.domain.key
+    Generate Local ModelRegistry Certificates    ${DOMAIN}    ${generate_certs_script}    ${cert_files}
+
+Generate Local ModelRegistry Certificates
+    [Documentation]    Generates ModelRegistry certificates using the generate_certs.sh script
+    [Arguments]    ${domain}    ${generate_certs_script}    ${cert_files}
+    Run Process    ${generate_certs_script}    ${domain}    stdout=PIPE    stderr=PIPE
+    Check Certificate Files Created    ${cert_files}
+
+Create Model Registry Secrets
+    [Documentation]    Create multiple generic secrets in the specified namespace with specified files
+    Create Generic Secret    ${NAMESPACE_ISTIO}    ${SECRET_PART_NAME_1}-credential
+    ...    certs/${SECRET_PART_NAME_1}.domain.key    certs/${SECRET_PART_NAME_1}.domain.crt    certs/domain.crt
+    Create Generic Secret    ${NAMESPACE_ISTIO}    ${SECRET_PART_NAME_2}-credential
+    ...    certs/${SECRET_PART_NAME_2}.domain.key    certs/${SECRET_PART_NAME_2}.domain.crt    certs/domain.crt
+    Create Generic Secret    ${NAMESPACE_MODEL_REGISTRY}  ${SECRET_PART_NAME_3}-credential
+    ...    certs/${SECRET_PART_NAME_3}.key           certs/${SECRET_PART_NAME_3}.crt     certs/domain.crt
+    Secret Should Exist      ${NAMESPACE_ISTIO}    ${SECRET_PART_NAME_1}-credential
+    Secret Should Exist      ${NAMESPACE_ISTIO}    ${SECRET_PART_NAME_2}-credential
+    Secret Should Exist      ${NAMESPACE_MODEL_REGISTRY}  ${SECRET_PART_NAME_3}-credential
+
+Secret Should Exist
+    [Documentation]    Check if the specified secret exists in the given namespace
+    [Arguments]    ${namespace}    ${secret_name}
+    ${output}=    Run Process    oc get secret ${secret_name} -n ${namespace}    shell=True
+    Should Contain    ${output.stdout}    ${secret_name}
```

Comment:
you could use `Resource Should Exist`  from `ods_ci/tests/Resources/OCP.resource`

Reply:
will be refactored in the future